legal-api/src/legal_api/services/filings/validations/common_validations.py

```diff
@@ -794,6 +795,20 @@ def validate_name_translation(filing_json: dict, filing_type: str) -> list:
 
     return msg
 
+def validate_officer_proprietor_replace(business: Business, filing_json: dict, filing_type) -> Optional[str]:
+    """Validate that sole proprietor is not being replaced with another sole proprietor."""
+    parties = filing_json["filing"][filing_type].get("parties", [])
+
+    for party in parties:
+        officer_identifier = party.get("officer", {}).get("identifier")
+        roles = party.get("roles", [])
+        has_proprietor_role = any(role.get("roleType").lower() == PartyRole.RoleTypes.PROPRIETOR.value for role in roles)
+        if business.legal_type == Business.LegalTypes.SOLE_PROP.value and has_proprietor_role:
+            existing_proprietor = Party.find_by_identifier(officer_identifier)
+            if existing_proprietor and existing_proprietor.identifier != officer_identifier:
```

Comment:
This is not correct. This code only consider the identifier in json.

Follow what we do in filer, use id to identify
Ref: https://github.com/bcgov/lear/blob/main/queue_services/business-filer/src/business_filer/filing_processors/change_of_registration.py#L110